app/src/main/java/edu/mit/mitmobile2/facilities/activity/PlaceActivity.java

```diff
@@ -0,0 +1,266 @@
+    private static final String NEARBY_LOCATIONS = "Nearby Locations";
+    private static final String BUILDINGS = "Buildings";
+    private static final String RESIDENCE = "Residences";
+    private static final String HIDDEN = "hidden=YES";
+    private static final String LEASED = "leased=YES";
+    private static final String NAME = "name";
+    private static final String PHONE = "phone";
+    private static final String EMAIL = "email";
+
+    @InjectView(R.id.place_list_view)
+    ListView placeListView;
+
+    private List<String> locationIdsbyCategory;
+    private String name;
+    private List<MITMapPlace> allLocations;
+    private List<MITMapPlace> places;
+    private PlaceAdapter placeAdapter;
+    private LocationCallback callback;
+    private List<String> hiddenLocationIds;
+    private List<FacilitiesPropertyOwner> propertyOwners;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_place);
+        ButterKnife.inject(this);
+
+        callback = (LocationCallback) this;
+        name = getIntent().getStringExtra(Constants.FACILITIES_CATEGORY_KEY);
+        setTitle(name);
+
+        places = new ArrayList<>();
+        fetchPlaces();
+        placeAdapter = new PlaceAdapter(this, places, callback);
+        placeListView.setAdapter(placeAdapter);
+    }
+
+    private void fetchPlaces() {
+        allLocations = new ArrayList<>();
+        MapManager.getMapPlaces(this, new Callback<ArrayList<MITMapPlace>>() {
+            @Override
+            public void success(ArrayList<MITMapPlace> places, Response response) {
+                for (MITMapPlace place : places) {
+                    if (name.equals(BUILDINGS) || name.equals(RESIDENCE)) {
+                        if (place.getBuildingNumber() == null) {
+                            place.parseNumber(place.getName());
+                        } else {
+                            place.parseNumber(place.getBuildingNumber());
+                        }
+                    } else {
+                        place.parseNumber(place.getName());
+                    }
+                    allLocations.add(place);
+                }
+
+                fetchLocationProperties();
+            }
+
+            @Override
+            public void failure(RetrofitError error) {
+                MitMobileApplication.bus.post(new OttoBusEvent.RetrofitFailureEvent(error));
+            }
+        });
+    }
+
+    private void fetchLocationProperties() {
+        hiddenLocationIds = new ArrayList<>();
+        propertyOwners = new ArrayList<>();
+        FacilitiesManager.getLocationProperties(this, new Callback<HashMap<String, HashMap<String, String>>>() {
+            @Override
+            public void success(HashMap<String, HashMap<String, String>> stringHashMapHashMap, Response response) {
+                Iterator iterator = stringHashMapHashMap.entrySet().iterator();
```

Comment:
Instead of pulling apart the json strings in here, you should use a custom deserializer in RetrofitManager to parse the json and output an object. There's a example in the bottom of the class file